src/freedesktop.rs

```diff
@@ -197,6 +197,32 @@ pub fn list() -> Result<Vec<TrashItem>, Error> {
     Ok(result)
 }
 
+pub fn is_empty() -> Result<bool, Error> {
+    let trash_folders = trash_folders()?;
+
+    if trash_folders.is_empty() {
+        return Ok(true);
+    }
+
+    for folder in trash_folders {
+        // We're only concerned if the trash contains any files
+        // Therefore, we only need to check if the bin itself is empty
+        let bin = folder.join("files");
+        match bin.read_dir() {
+            Ok(mut entries) => {
+                if let Some(Ok(_)) = entries.next() {
+                    return Ok(false);
+                }
+            }
+            Err(e) => {
+                warn!("The trash files folder {:?} could not be read. Error was {:?}", bin, e);
```

Comment:
This is an interesting issue - should everything be an error if one folder can't be read?
Right now it assume they are empty even if all of them can't be read.

Unfortunately I don't know if it's common to try to access 'invalid' folders, so I guess we can try to run with this for a while.